docs/apis/plugintypes/format/_examples/lib.php

```diff
@@ -0,0 +1,104 @@
+
+class format_pluginname extends core_courseformat\base {
+
+    /**
+     * Returns true if this course format uses sections.
+     *
+     * @return bool
+     */
+    public function uses_sections() {
+        return true;
+    }
+
+    public function uses_indentation(): bool {
+        return false;
+    }
+
+    public function uses_course_index() {
+        return true;
+    }
+
+    /**
+     * Returns the information about the ajax support in the given source format.
+     *
+     * The returned object's property (boolean)capable indicates that
+     * the course format supports Moodle course ajax features.
+     *
+     * @return stdClass
+     */
+    public function supports_ajax() {
+        $ajaxsupport = new stdClass();
+        $ajaxsupport->capable = true;
+        return $ajaxsupport;
+    }
+
+    public function supports_components() {
+        return true;
+    }
+
+    /**
+     * Whether this format allows to delete sections.
+     *
+     * Do not call this function directly, instead use {@link course_can_delete_section()}
+     *
+     * @param int|stdClass|section_info $section
+     * @return bool
+     */
+    public function can_delete_section($section) {
+        return true;
+    }
+
+    /**
+     * Indicates whether the course format supports the creation of a news forum.
+     *
+     * @return bool
+     */
+    public function supports_news() {
+        return true;
+    }
+
+    /**
+     * Returns the display name of the given section that the course prefers.
+     *
+     * This method is required for inplace seciton name editor.
+     *
+     * @param int|stdClass $section Section object from database or just field section.section
+     * @return string Display name that the course format prefers, e.g. "Topic 2"
+     */
+    public function get_section_name($section) {
+        $section = $this->get_section($section);
+        if ((string)$section->name !== '') {
+            return format_string(
+                $section->name,
+                true,
+                ['context' => context_course::instance($this->courseid)]
+            );
+        } else {
+            return $this->get_default_section_name($section);
+        }
+    }
+}
+
+/**
+ * Implements callback inplace_editable() allowing to edit values in-place.
+ *
+ * This method is required for inplace seciton name editor.
+ *
+ * @param string $itemtype
+ * @param int $itemid
+ * @param mixed $newvalue
+ * @return inplace_editable
+ */
+function format_pluginname_inplace_editable($itemtype, $itemid, $newvalue) {
```

Comment:
Depends on the definition of relevant. It is not mandatory but it is the kind of integration that makes developers lose many hours because it is not documented. Having it in the example makes everything smoother for new developers because they will end up with a fully functional format.

Reply:
Okay +1